src/Ocelot/DownstreamRouteFinder/UrlMatcher/UrlPathPlaceholderNameAndValueFinder.cs

```diff
@@ -80,6 +82,33 @@ private static List<Group> FindGroups(string path, string query, string template
         match = Regex.Match($"{testedPath}/", regexPattern);
         return match.Groups.Cast<Group>().Skip(1).ToList();
     }
+    
+    /// <summary>
+    /// The placeholders that are not placed at the end of the template
+    /// are delimited by forward slashes, only the last one, the catch-all can match
+    /// more segments.
+    /// </summary>
+    /// <param name="escapedTemplate">The escaped path template.</param>
+    /// <returns>The pattern for values replacement.</returns>
+    private static string GenerateRegexPattern(string escapedTemplate)
+    {
+        // first we count the matches
+        var placeHoldersCountMatch = RegexPlaceholders().Matches(escapedTemplate);
+        int placeHoldersCount = placeHoldersCountMatch.Count;
+        
+        int index = 0;
+        
+        // we know that the replace process will be started from the beginning of the url
+        // so we can use a simple counter to determine the last placeholder
+        var regexPattern = $@"^{RegexPlaceholders().Replace(escapedTemplate, match =>
+        {
+            var groupName = match.Groups[1].Value;
+            index++;
+            return index == placeHoldersCount ? $"(?<{groupName}>[^&]*)" : $"(?<{groupName}>[^/|&]*)"; 
```

Comment:
So, this `[^/|&]` negative matching group filters out all slashes `/` after closing brace `}`, before catch all placeholder, right?
And, the last RX-match, for the last Catch-All, just accepts uncounted number of slashes before query string, right?
Query string processing is quite unclear... What's the final generated pattern?